web/elm/src/Message/Message.elm

```diff
@@ -14,72 +14,63 @@ import StrictEvents
 
 type Message
     = -- Top Bar
-      LogIn
-    | LogOut
-    | FilterMsg String
+      FilterMsg String
     | FocusMsg
     | BlurMsg
-    | ToggleUserMenu
-    | ShowSearchInput
-    | TogglePipelinePaused Concourse.PipelineIdentifier Bool
       -- Pipeline
     | ToggleGroup Concourse.PipelineGroup
     | SetGroups (List String)
-      -- Fly Success
-    | CopyToken
       -- Dashboard
     | DragStart String Int
     | DragOver String Int
     | DragEnd
     | Tooltip String String
     | TooltipHd String String
       -- Resource
-    | LoadPage Page
-    | ExpandVersionedResource VersionId
-    | PinVersion VersionId
-    | UnpinVersion
-    | ToggleVersion VersionToggleAction VersionId
-    | CheckRequested Bool
     | EditComment String
-    | SaveComment String
     | FocusTextArea
     | BlurTextArea
-      -- Job
-    | TriggerBuildJob
-    | TogglePaused
       -- Build
-    | SwitchToBuild Concourse.Build
-    | TriggerBuild (Maybe Concourse.JobIdentifier)
-    | AbortBuild Int
     | ScrollBuilds StrictEvents.MouseWheelEvent
     | RevealCurrentBuildInHistory
-    | ToggleStep String
-    | SwitchTab String Int
     | SetHighlight String Int
     | ExtendHighlight String Int
       -- common
-    | Hover (Maybe Hoverable)
+    | Hover (Maybe DomID)
+    | Click DomID
     | GoToRoute Routes.Route
 
 
-type Hoverable
+type DomID
```

Comment:
...Downside - this sucker is getting pretty gross. Grrrrr. Is just by string ids better or worse, do you think?

Reply:
I'd like to see the massive-enum approach through. We'll see if the compiler errors wind up feeling helpful or irritating.

Reply:
Legit. Certainly we'll be in a better place to understand the tradeoffs fully when we have *everything* in it, for sure.